src/guided_backprop.py

```diff
@@ -62,6 +59,9 @@ def relu_forward_hook_function(module, ten_in, ten_out):
                 module.register_forward_hook(relu_forward_hook_function)
 
     def generate_gradients(self, input_image, target_class):
+        # Register input hook
+        input_image.requires_grad = True
+        self.hook_input(input_image)
```

Comment:
A small tip on this. I personally think it is better to use a deep copy of the input_image to ensure reproducibility. Otherwise, we might have an increasing number of hooks for the exact input_image if we call the function multi times.

Reply:
Good idea, thanks!
I've updated the code adding the copy of the tensor.